src/test/java/de/tum/cit/aet/artemis/exercise/repository/ExerciseTestRepository.java

```diff
@@ -0,0 +1,31 @@
+package de.tum.cit.aet.artemis.exercise.repository;
+
+import java.util.List;
+import java.util.Set;
+
+import org.springframework.context.annotation.Primary;
+import org.springframework.data.jpa.repository.EntityGraph;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+
+import de.tum.cit.aet.artemis.exercise.domain.Exercise;
+
+/**
+ * Spring Data JPA repository for the Exercise entity for Tests.
+ */
+@Primary
+@Repository
+public interface ExerciseTestRepository extends ExerciseRepository {
```

Comment:
Seems like introducing a test repository has caused the architecture tests to fail with 321 violations. How do we want to proceed here?

----

```
  java.lang.AssertionError: Architecture Violation [Priority: MEDIUM] - Rule 'classes should not use repositories with subclasses, because Test Repositories should be used over production repositories, if such a repository exist.' was violated (321 times):
  Test class de.tum.cit.aet.artemis.assessment.AssessmentComplaintIntegrationTest uses repository de.tum.cit.aet.artemis.exercise.repository.ExerciseRepository which has subclasses: [JavaClass{name='de.tum.cit.aet.artemis.exercise.repository.ExerciseTestRepository'}]
...
```